### PR TITLE
Make all pbuilder hooks static

### PR DESCRIPTION
**Pull request by ekohl**
It also moves F60addforemanrepo to a static hook by using `$DISTRIBUTION` which according to man pbuilder should be present in every hook.

It also splits all hooks to separate files. Rather than changing the content depending on the preferences, certain files are deployed or not deployed. The whole directory is now managed and unmanaged files are purged. This is possible since Jenkins jobs no longer modify hooks.

Includes https://github.com/theforeman/foreman-infra/pull/1644

### CODE REVIEW

File: puppet/modules/slave/manifests/pbuilder_setup.pp [hunk 1]
```diff
@@ -34,34 +34,29 @@
...
   }
 
-  file { "/etc/pbuilder/${name}/hooks/F60addforemanrepo":
-    ensure  => $ensure,
-    mode    => '0775',
-    content => template('slave/pbuilder_F60addforemanrepo.erb'),
+  file { "/etc/pbuilder/${name}/hooks":
+    ensure => directory,
+    mode   => '0775',
+    purge  => true,
```
Comment (ekohl) [inline puppet/modules/slave/manifests/pbuilder_setup.pp:40]: Note: I haven't checked if there are unmanaged hooks right now.

File: puppet/modules/slave/manifests/pbuilder_setup.pp [hunk 2]
```diff
@@ -34,34 +34,29 @@
...
-    mode    => '0775',
-    content => template('slave/pbuilder_f70.erb'),
-  }
+  $hooks = [
+    'C10foremanlog',
+    'D80no-man-db-rebuild',
+    'F60addforemanrepo',
+    if $backports { 'F65-add-backport-repos' },
+    if $nodesource { 'F66-add-nodesource-repos' },
+    if $puppetlabs { 'F66-add-puppet-repos' },
```
Comment (ekohl) [inline puppet/modules/slave/manifests/pbuilder_setup.pp:49]: I must admit I haven't tested this just yet, but in theory it should work.

**Comment by ekohl** [comment]
Bleh, looks like this doesn't work since pbuilder already manages the directory and doesn't have a purge parameter.

**Comment by ekohl** [comment]
Includes https://github.com/theforeman/foreman-infra/pull/1650

**Review by ehelms — APPROVED** [review]
This looks good to me, I am not the Debian expert though. What sets $DISTRIBUTION?

**Comment by ekohl** [comment]
> What sets $DISTRIBUTION?

https://github.com/theforeman/foreman-infra/pull/1650#issuecomment-911685989

**Review by evgeni — APPROVED** [review]

**Comment by ekohl** [comment]
Rebased

File: puppet/spec/classes/slave_spec.rb [hunk 3]
```diff
@@ -17,6 +17,13 @@
         else
           it { is_expected.to contain_users__account('jenkins').with_sudo('') }
         end
+
+        if facts[:osfamily] == 'Debian'
+          it { is_expected.to contain_file('/etc/pbuilder/buster64/hooks/F60addforemanrepos') }
```
Comment (evgeni) [inline puppet/spec/classes/slave_spec.rb:22]:
```suggestion
          it { is_expected.to contain_file('/etc/pbuilder/buster64/hooks/F60addforemanrepo') }
```

File: puppet/spec/classes/slave_spec.rb [hunk 4]
```diff
@@ -17,6 +17,13 @@
...
+          it { is_expected.not_to contain_file('/etc/pbuilder/buster64/hooks/F65-add-backport-repos') }
```
Comment (evgeni) [inline puppet/spec/classes/slave_spec.rb:23]: this fails?

Comment (ekohl) [reply puppet/spec/classes/slave_spec.rb]: Correct. I changed the code to ensure absent. Now the tests expect the same.

**Comment by ekohl** [comment]
Updated, now with green tests.

**Review by evgeni — APPROVED** [review]

**Pull request merged**